src/Libraries/Microsoft.Extensions.Diagnostics.ResourceMonitoring/Windows/Interop/MemoryInfo.Native.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft Corporation. All Rights Reserved.
+
+using System.Runtime.InteropServices;
+using Microsoft.Extensions.Diagnostics.ResourceMonitoring.Windows.Interop;
+
+#if NET8_0_OR_GREATER
+using DllImportAttr = System.Runtime.InteropServices.LibraryImportAttribute; // We trigger source-gen on .NET 7 and above
+#else
+using DllImportAttr = System.Runtime.InteropServices.DllImportAttribute;
+#endif
+
+namespace Microsoft.Extensions.Diagnostics.ResourceMonitoring.Windows.Interop;
+
+internal sealed partial class MemoryInfo
+{
+    private static partial class SafeNativeMethods
+    {
+        /// <summary>
+        /// GlobalMemoryStatusEx.
+        /// </summary>
+        /// <param name="memoryStatus">Memory Status structure.</param>
+        /// <returns>Success or failure.</returns>
+        [DllImportAttr("kernel32.dll", SetLastError = true)]
+        [DefaultDllImportSearchPaths(DllImportSearchPath.System32)]
+        [return: MarshalAs(UnmanagedType.Bool)]
```

Comment:
This should be turned into a blittable `BOOL` to avoid marshaling which has perf overhead
https://github.com/dotnet/runtime/blob/31733b9a35185785427bac69ef80a4eb56b727c2/src/libraries/Common/src/Interop/Windows/Kernel32/Interop.GlobalMemoryStatusEx.cs

Reply:
Added